pkg/factory/setk8sdefaults_test.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2019 The Kubernetes Authors.
+// Copyright 2019-2021 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+package factory
+
+import (
+	"testing"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSetKubernetesDefaults(t *testing.T) {
+	t.Run("sets default GroupVersion", func(t *testing.T) {
+		config := &rest.Config{}
+		err := SetKubernetesDefaults(config)
+		require.NoError(t, err)
+		require.NotNil(t, config.GroupVersion)
+		require.Equal(t, schema.GroupVersion{Group: "", Version: "v1"}, *config.GroupVersion)
+	})
+
+	t.Run("sets default APIPath", func(t *testing.T) {
+		config := &rest.Config{}
+		err := SetKubernetesDefaults(config)
+		require.NoError(t, err)
+		require.Equal(t, "/api", config.APIPath)
+	})
+
+	t.Run("does not override existing APIPath", func(t *testing.T) {
+		config := &rest.Config{APIPath: "/custom"}
+		err := SetKubernetesDefaults(config)
+		require.NoError(t, err)
+		require.Equal(t, "/custom", config.APIPath)
+	})
+
+	t.Run("sets default NegotiatedSerializer", func(t *testing.T) {
```

Comment:
Add another test which checks if it does not override an existing `NegotiatedSerializer`